jaypeasee/rancid-tomatillos
# Feature/search bar

# Pull Request Template

#### What’s this PR do?
This PR:
1. creates a search bar to filter movies in the gallery view by movie title.
* Creates functionality to make the search non case sensitive,
* Clears the inputs after submission
* Makes it easy to return to all movie (non filtered gallery view)
* Styles the search bar in all breakpoints
* Unit tests the searchBar elements and function.
* Integration tests the results of the search bar use
2. Cleans up rogue semi-colons, orders imports, and spaces deconstructions & JSX in all files.


#### Where should the reviewer start?
* Start in `App.js` to see how the `<Search Bar/>` component is rendered and how data is passed down to it.
* Review `SearchBar.js` to see how the form is handled and actions are passed back up to the `<App />` component.
* Review `Searchbar.scss` to see how the component is styled.
* Review all of `Searchbar.test.js` to see how the unit tests are set up.
* Review the last test in `App.test.js` to see how the integration test is setup.


#### How should this be manually tested?
1. Play with the search bar. Enter non-case-sensitive movie titles and check the results.
1. Return to all movies by clicking one of the home buttons in the nav bar and the search button again when the input is empty.
1. Run `npm test` to see that all the tests are passing.

#### Any background context you want to provide? 
This completes a choose-your-own-adventure initiative we had for iteration 5.

#### Screenshots (if appropriate): 
![Screen Shot 2020-12-12 at 11 31 10 AM](https://user-images.githubusercontent.com/59029768/101992002-8bd49a00-3c6d-11eb-883f-e2592b70d3ca.png)
